BCStudentSoftwareDevTeam/celts
# Profile UI for CCE Minor Experiences.

## Issue Description

Fixes #1152 
- We needed to create a new page where we could create Summer Experiences with all of the information CELTS needs instead of just one text box.

## Changes
- Created a new tab system for navigating between Sustained Engagements, Summer Experience, and Other Engagements.
- Created a new form for submitting a Summer Experience.
- Created a new model for summerExperience.
- Fixed up and made the Request Other Engagement page accessible.
- Added phone number validation to the Company/Org part of the form.
- Modularized the HTML for reusability.
- Dynamically populated the years drop downs.
- Wrote tests for the new createSummerExperience function.

## Testing
- After setting up environment and resetting the databases, run tests with `./tests/run_tests.sh`
- Run the application using `flask run`.
- Click on the Minor Management tab in the Admin sidebar.
- Click on any of the names under the CCE Minor Progress table (we did most of our testing on Ebenezer Ayisi.
- Select the Summer Experience tab to open up the form.
- Check the styling and then fill out all the information and submit.
- Once kicked back to the Sustained Community Engagements page, return to Summer Experience and attempt to fill out the form but make sure to select **Other** for the 3rd to last question and **no** for the final question to verify the conditional check boxes are working.
- Check the database with `mysql -u root -proot` ->  `use celts` -> `select * from summerexperience;` to make sure the data you just submitted is all there.
- Merge our PR :)
